src/runtime_src/core/pcie/linux/pcidev.cpp

```diff
@@ -467,57 +428,43 @@ get_subdev_path(const std::string& subdev, uint idx)
   std::string path("/dev/xfpga/");
 
   path += subdev;
-  path += is_mgmt() ? ".m" : ".u";
+  path += is_mgmt ? ".m" : ".u";
   //if the domain number is big, the shift overflows, hence need to cast
   uint32_t dom = static_cast<uint32_t>(domain);
-  path += std::to_string( (dom<<16)+ (bus<<8) + (dev<<3) + func);
+  path += std::to_string( (dom<<16)+ (bus<<8) + (dev_no<<3) + func);
   path += "." + std::to_string(idx);
   return path;
 }
 
 int
-pci_device::
-open(const std::string& subdev, uint32_t idx, int flag)
+dev::
+open(const std::string& subdev, uint32_t idx, int flag) const
 {
-  if (is_mgmt() && !::is_admin())
+  if (is_mgmt && !::is_admin())
     throw std::runtime_error("Root privileges required");
 
   std::string devfs = get_subdev_path(subdev, idx);
   return ::open(devfs.c_str(), flag);
 }
 
 int
-pci_device::
-open(const std::string& subdev, int flag)
+dev::
+open(const std::string& subdev, int flag) const
 {
   return open(subdev, 0, flag);
 }
 
-pci_device::
-pci_device(const std::string& drv_name, const std::string& sysfs) : sysfs_name(sysfs)
+dev::
+dev(const drv& driver, const std::string& sysfs) : sysfs_name(sysfs)
```

Comment:
The constructor should be initializing all of the member variables.

Reply:
add "m_user_bar_map = reinterpret_cast<char *>(MAP_FAILED);" in the constructor.